src/components/notebook-utils.js

```diff
@@ -72,22 +72,43 @@ export const analysisNameInput = ({ inputProps, ...props }) => h(ValidatedInput,
 })
 
 export const tools = {
-  RStudio: { label: 'RStudio', ext: 'Rmd', imageIds: ['RStudio'], defaultImageId: 'RStudio' },
-  Jupyter: { label: 'Jupyter', ext: 'ipynb', imageIds: ['terra-jupyter-bioconductor', 'terra-jupyter-bioconductor_legacy', 'terra-jupyter-hail', 'terra-jupyter-python', 'terra-jupyter-gatk', 'Pegasus', 'terra-jupyter-gatk_legacy'], defaultImageId: 'terra-jupyter-gatk', isLaunchUnsupported: true },
+  RStudio: { label: 'RStudio', ext: ['Rmd', 'R'], imageIds: ['RStudio'], defaultImageId: 'RStudio', defaultExt: 'Rmd' },
+  Jupyter: { label: 'Jupyter', ext: ['ipynb'], imageIds: ['terra-jupyter-bioconductor', 'terra-jupyter-bioconductor_legacy', 'terra-jupyter-hail', 'terra-jupyter-python', 'terra-jupyter-gatk', 'Pegasus', 'terra-jupyter-gatk_legacy'], defaultImageId: 'terra-jupyter-gatk', isLaunchUnsupported: true, defaultExt: 'ipynb' },
   jupyterTerminal: { label: 'terminal' },
   spark: { label: 'spark' },
   Galaxy: { label: 'Galaxy', appType: 'GALAXY' },
   Cromwell: { label: 'Cromwell', appType: 'CROMWELL', isAppHidden: !isCromwellAppVisible(), isPauseUnsupported: true },
   Azure: { label: 'Azure', isAzureCompatible: true, isPauseUnsupported: true, isLaunchUnsupported: false }
 }
 
+export const toolExtensionDisplay = {
+  RStudio: [
+    { label: 'R Markdown (.Rmd)', value: 'Rmd' },
+    { label: 'R Script (.R)', value: 'R' }
+  ],
+  Jupyter: [{ label: 'IPython Notebook (.ipynb)', value: 'ipynb' }]
+}
+
+export const getPatternFromTool = toolLabel => Utils.switchCase(toolLabel,
+  [tools.RStudio.label, () => '.*\\.R(md)'],
+  [tools.Jupyter.label, () => '.*\\.ipynb']
+)
+
+export const addExtensionToNotebook = name => `${name}.${tools.Jupyter.defaultExt}`
```

Comment:
I like that this is a function now instead of having `ipynb`s and string interpolation thrown everywhere